nvflare/apis/impl/job_def_manager.py

```diff
@@ -52,25 +59,53 @@ class _AllJobsFilter(_JobFilter):
     def __init__(self):
         self.result = []
 
-    def filter_job(self, meta: dict):
+    def filter_job(self, meta: dict, job_id: str):
         self.result.append(job_from_meta(meta))
         return True
 
 
 class _ReviewerFilter(_JobFilter):
-    def __init__(self, reviewer_name, fl_ctx: FLContext):
+    def __init__(self, reviewer_name):
         """Not used yet, for use in future implementations."""
         self.result = []
         self.reviewer_name = reviewer_name
 
-    def filter_job(self, meta: dict):
+    def filter_job(self, meta: dict, job_id: str):
         approvals = meta.get(JobMetaKey.APPROVALS)
         if not approvals or self.reviewer_name not in approvals:
             self.result.append(job_from_meta(meta))
         return True
 
 
-# TODO:: use try block around storage calls
+class _ScheduleJobFilter(_JobFilter):
+
+    """
+    This filter is optimized for selecting jobs to schedule since it is used so frequently (every 1 sec).
+    """
+
+    def __init__(self):
+        self.result = []
+        self.skipped_jobs = {}  # job id => bool for quick lookup.
```

Comment:
One question here, if the server is going to run forever
Will this self.skipped_jobs keeps increasing?
Do we ever reset it?

Reply:
Right. I'm working on a better version that uses timestamp instead.